java/README.rst

```diff
@@ -0,0 +1,202 @@
+        .Builder(tracer)
+        .withStreaming()
+        .withVerbosity()
+        .withOperationName("service-name")
+        .withTracedAttributes(ServerRequestAttribute.HEADERS, 
+            ServerRequestAttribute.METHOD_TYPE)
+        .build();
+
+==============
+Client Tracing
+==============
+
+A `ClientTracingInterceptor` also has default settings, which you can override by creating it using a `ServerTracingInterceptor.Builder`.
+
+- `withOperationName(String operationName)`: Sets the operation name for all spans created for this intercepted client. Default is the name of the RPC method.
+- `withStreaming()`: Logs to the client span whenever a message is sent or a response is received.
+- `withVerbosity()`: Logs to the client span additional events, such as call started, message sent, half close (client finished sending messages), response received, and call complete. Default only logs if a call is cancelled.  
+- `withTracedAttributes(ClientRequestAttribute... attrs)`: Sets tags on the client span in case you want to track information about the RPC call. See ClientRequestAttribute.java for a list of traceable request attributes.
+
+**Example**:
+
+.. code-block:: java
+
+    ClientTracingInterceptor tracingInterceptor = new ClientTracingInterceptor
+        .Builder(tracer)
+        .withStreaming()
+        .withVerbosity()
+        .withOperationName("client-name")
+        .withTracingAttributes(ClientRequestAttribute.ALL_CALL_OPTIONS,
+            ClientRequestAttribute.HEADERS)
+        .build();
+
+====================
+Current Span Context
+====================
+
+In your server request handler, you can access the current active span for that request by calling
+
+.. code-block:: java
+
+    Span span = OpenTracingContextKey.activeSpan();
+
+This is useful if you want to manually set tags on the span, log important events, or create a new child span for internal units of work. You can actually use this key to wrap these internal units of work with a new context that has a user-defined active span.
+
+For example:
+
+.. code-block:: java
+
+    Tracer tracer = someOpenTracingTracer;
+
+    // some unit of internal work that you want to trace
+    Runnable internalWork = new Runnable() {
+        public void run() {
+            // do some work
+        }
+    } 
+
+    // get the active span 
+    Span activeSpan = OpenTracingContextKey.activeSpan();
+
+    // create a child span to represent the internal work
+    Span childSpan = tracer.buildSpan("internal-work").asChildOf(activeSpan).start();
+
+    // define a new context with childSpan as the active span
+    Context contextWithChildSpan = Context.current().withValue(OpenTracingContextKey.get(), childSpan);
+
+    // wrap the internal work with this new context so that the active span in the scope 
+    // of internalWork is childSpan
+    Runnable tracedInternalWork = contextWithChildSpan.wrap(internalWork)
```

Comment:
we should keep this example, but we should also create some utilities to make this idiom easier... probably an extension of the `Runnable` interface that grabs the current `activeSpan` from the context and does most of the wrapping internally.